### PR TITLE
Bug 1224063 - Refactor System to use modern L10n API, part 1

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1224063

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c7e7d59d8755baee6f4fc543a4922c15e840a8fc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=654089cda5844be22fc29473e78d3bfa32122fe8

File: apps/system/js/app_install_manager.js [hunk 1]
```diff
@@ -243,22 +242,28 @@
       var id = detail.id;
 
       if (manifest.size) {
+        this.size.removeAttribute('data-l10n-id');
         this.size.textContent = this.humanizeSize(manifest.size);
       } else {
-        this.size.textContent = _('size-unknown');
+        this.size.setAttribute('data-l10n-id', 'size-unknown');
```
Comment (gasolin) [inline apps/system/js/app_install_manager.js]: will it be translated on the fly by mozl10n when new data-l10n-id attribute is appended? or we could use navigator.mozL10n.setAttributes to set string for safety

File: apps/system/js/migrators/settings_migrator.js [hunk 2]
```diff
@@ -9,7 +9,6 @@
    * @class SettingsMigrator
    */
   var SettingsMigrator = function SettingsMigrator() {
-    this._kLocaleTime = 'locale.hour12';
```
Comment (gasolin) [inline apps/system/js/migrators/settings_migrator.js]: \o/

File: apps/system/js/app_install_manager.js (hunk 1 above)
Comment (zbraniecki) [inline apps/system/js/app_install_manager.js]: L10n.js uses `MutationObserver` to observe any changes to `data-l10n-id` and `data-l10n-args` and translates affected nodes.

**Pull request merged**